packages/runtime/src/lib/constants.ts

```diff
@@ -307,22 +307,94 @@ AssetParamMap[6] = { ...AssetParamMap[5] };
 AssetParamMap[7] = { ...AssetParamMap[6] };
 AssetParamMap[8] = { ...AssetParamMap[7] };
 
+export enum AssetParamGetField {
+	AssetTotal = "AssetTotal",
+	AssetDecimals = "AssetDecimals",
+	AssetDefaultFrozen = "AssetDefaultFrozen",
+	AssetUnitName = "AssetUnitName",
+	AssetName = "AssetName",
+	AssetURL = "AssetURL",
+	AssetMetadataHash = "AssetMetadataHash",
+	AssetManager = "AssetManager",
+	AssetReserve = "AssetReserve",
+	AssetFreeze = "AssetFreeze",
+	AssetClawback = "AssetClawback",
+	AssetCreator = "AssetCreator",
+}
+
+export enum AppParamField {
+	AppApprovalProgram = "AppApprovalProgram",
+	AppClearStateProgram = "AppClearStateProgram",
+	AppGlobalNumUint = "AppGlobalNumUint",
+	AppGlobalNumByteSlice = "AppGlobalNumByteSlice",
+	AppLocalNumUint = "AppLocalNumUint",
+	AppLocalNumByteSlice = "AppLocalNumByteSlice",
+	AppExtraProgramPages = "AppExtraProgramPages",
+	AppCreator = "AppCreator",
+	AppAddress = "AppAddress",
+}
+
+export enum TxnRefFields {
+	ApplicationArgs = "ApplicationArgs",
+	ConfigAssetTotal = "ConfigAssetTotal",
+	ConfigAssetDecimals = "ConfigAssetDecimals",
+	ConfigAssetDefaultFrozen = "ConfigAssetDefaultFrozen",
+	ConfigAssetUnitName = "ConfigAssetUnitName",
+	ConfigAssetName = "ConfigAssetName",
+	ConfigAssetURL = "ConfigAssetURL",
+	ConfigAssetMetadataHash = "ConfigAssetMetadataHash",
+	ConfigAssetManager = "ConfigAssetManager",
+	ConfigAssetReserve = "ConfigAssetReserve",
+	ConfigAssetFreeze = "ConfigAssetFreeze",
+	ConfigAssetClawback = "ConfigAssetClawback",
+	CreatedAssetID = "CreatedAssetID",
+	CreatedApplicationID = "CreatedApplicationID",
+	FirstValidTime = "FirstValidTime",
+	FreezeAssetAccount = "FreezeAssetAccount",
+	FreezeAssetFrozen = "FreezeAssetFrozen",
+	Global = "Global",
+	GlobalNumUint = "GlobalNumUint",
+	GroupIndex = "GroupIndex",
+	GlobalNumByteSlice = "GlobalNumByteSlice",
+	LocalNumByteSlice = "LocalNumByteSlice",
+	LocalNumUint = "LocalNumUint",
+	NumAppArgs = "NumAppArgs",
+	NumAccounts = "NumAccounts",
+	NumAssets = "NumAssets",
+	NumApplications = "NumApplications",
+	NumApprovalProgramPages = "NumApprovalProgramPages",
+	NumClearStateProgramPages = "NumClearStateProgramPages",
+	NumLogs = "NumLogs",
+	OnCompletion = "OnCompletion",
+	TxID = "TxID",
+}
+
+export enum TxnaField {
+	Accounts = "Accounts",
```

Comment:
I think these can also be used with regular txn/itxn opcodes

Reply:
Done